polkadot/node/subsystem-util/src/backing_implicit_view.rs

```diff
@@ -265,30 +291,91 @@ pub enum BlockHeaderUnavailableReason {
 struct FetchSummary {
 	minimum_ancestor_number: BlockNumber,
 	leaf_number: BlockNumber,
-	relevant_paras: Vec<ParaId>,
 }
 
-async fn fetch_fresh_leaf_and_insert_ancestry<Sender>(
+// Request the min relay parents from prospective-parachains.
+async fn prospective_parachains_min_relay_parents<
+	Sender: SubsystemSender<ProspectiveParachainsMessage>,
+>(
 	leaf_hash: Hash,
-	block_info_storage: &mut HashMap<Hash, BlockInfo>,
 	sender: &mut Sender,
-) -> Result<FetchSummary, FetchError>
+) -> Result<Vec<(ParaId, BlockNumber)>, FetchError> {
+	let (tx, rx) = oneshot::channel();
+	sender
+		.send_message(ProspectiveParachainsMessage::GetMinimumRelayParents(leaf_hash, tx))
+		.await;
+
+	rx.await.map_err(|_| FetchError::ProspectiveParachainsUnavailable)
+}
+
+// Request the min relay parent for a single para, directly using ChainApi.
```

Comment:
This seem to return the `leaf_number` - `allowed_ancestry_len` no matter what `para_id`. Don't we have to constrain this to be the minimum between this value and the best backable parachain block relay parent ?

Reply:
This would complicate the code for collators, since they'll no longer have the prospective-parachains subsystem to track the backed candidates. Also, they don't receive backing statements. I guess we could use the persisted_validation_data runtime API which would return the relay chain block height of the last backed or included candidate.

I wouldn't bother with this for now, as it preserves the existing behaviour (which doesn't seem to cause issues) and the collator should anyway be smart enough to not produce collations that move the relay parents backwards.

Reply:
the name of the function though is not really correct

Reply:
changed the function name